### ISSUE TITLE
monitor command breaks due to ReadTimeout

### ISSUE THREAD

**Issue by stlava** [redis/go-redis#3079]
## Expected Behavior

Monitor command should work.

## Current Behavior

If you do not set a large `ReadTimeout` on the client when using the monitor command then the socket timeout is hit but that error is never propagated back in any way so the channel listener is left in a blocked state.

## Possible Solution

Either:
1. Update docs to inform users that they are responsible for detecting timeouts when listening to the channel
2. A `ReadTimeout` (or any other error) should be propagated back to user via closing the monitor channel

## Steps to Reproduce

1. Start client
```
	client := redis.NewClient(&redis.Options{
		Addr: "127.0.0.1:6379",
		// ReadTimeout: 1 * time.Minute,
	})

	stream := make(chan string, 1000)
	monitorCmd := client.Monitor(ctx, stream)
	monitorCmd.Start()

	for {
		var msg string

		select {
		case <-ctx.Done():
			fmt.Println("ctx done")
		case msg = <-stream:
			fmt.Println("message")
		}

		fmt.Println(msg)
	}
```

2. Wait a second and then via cli perform any redis operation
3. Check output from client

**Comment by Stellacska11** [comment]
![Image](https://github.com/user-attachments/assets/5caf5006-0205-4304-9c4a-123e99e4a31f)

**Comment by ndyakov** [comment]
@stlava is this still a problem with the latest versions?

**Comment by ankit-songara** [comment]
@ndyakov Yes, this still reproduces on latest master (fbdeae2 base). The root cause is that the error from rd.Peek(1) in MonitorCmd.readMonitor is discarded, so after a read timeout the goroutine spins forever re-peeking a dead connection, cmd.Err() stays nil, and anything blocked on the channel hangs indefinitely.

I've opened a PR that propagates the error, cancels the context, and closes the channel so listeners wake up and can check cmd.Err(). It includes a regression test using a fake in-process server (no real Redis needed) that fails on master and passes with the fix.
